lib/puppet/type/cpan.rb

```diff
@@ -21,6 +21,27 @@
         provider.update
       end
     end
+
+    def insync?(is)
```

Comment:
What is the purpose of this function ?
It doesn't seem to be called anywhere

Reply:
This is an override of the method within property.rb and is called during the evaluation.
( See: https://github.com/puppetlabs/puppet/blob/main/lib/puppet/property.rb#L307 )

This is commonly done for complex comparisons, however in this case it's used to short circuit the default handling and make it simpler.

Reply:
@faxm0dem -- Is there anything additional I can do to assist?

I realized I didn't answer your first question, "What is the purpose of this function?"
It is what performs the test to determine if the module is "in sync".
It will create the provider (cpan) instance if needed, then will compare based on the requested ensure type to determine if the provider needs to complete an installation or not.

Reply:
:+1: